std/algorithm.d

```diff
@@ -6626,9 +6626,9 @@ struct Until(alias pred, Range, Sentinel) if (isInputRange!Range)
     private bool predSatisfied()
     {
         static if (is(Sentinel == void))
-            return unaryFun!pred(_input.front);
+            return cast(bool) unaryFun!pred(_input.front);
         else
-            return startsWith!pred(_input, _sentinel);
+            return cast(bool) startsWith!pred(_input, _sentinel);
```

Comment:
If there is a difference then it's most likely a bug.  IIRC even dmd's backend can fix `e ? true : false`.